src/main/java/org/manifold/compiler/front/ExpressionGraph.java

```diff
@@ -279,7 +279,19 @@ public void addFunctionExpressionGraph(ExpressionGraph subGraph,
     exprVertexMap.put(subGraphInput, inputVertex);
 
     // Connect the function output vertex to the main graph
-    ExpressionEdge outputEdge = new ExpressionEdge(exprVertexMap.get(subGraphOutput), mainGraphOutput.getTarget());
+    ExpressionVertex outputVertex = exprVertexMap.get(subGraphOutput);
+    if (((TupleValueVertex) subGraphOutput).getValueEdges().size() == 1) {
```

Comment:
It's been a while, but I think it's because we create that tuple output. So it's guaranteed to be a tuple since we made it that way. Potentially we can change it at the function signature level, I'll check that